s3/s3.go

```diff
@@ -94,6 +106,16 @@ func NewS3Client(opts S3ClientOpts) (S3Client, error) {
 		} else {
 			minioClient, err = minio.New(s3cli.Endpoint, s3cli.AccessKey, s3cli.SecretKey, s3cli.Secure)
 		}
+	} else if os.Getenv("AWS_ROLE_ARN") != "" && os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE") != "" {
+		log.Infof("Creating minio client %s using WebIdentityCredentials credentials", os.Getenv("AWS_ROLE_ARN"))
+		cred, err := GetWebIdentityCredentials(opts)
+		if err != nil {
+			return nil, errors.WithStack(err)
```

Comment:
How come you’re wrapping the error here? I don’t really see this behavior elsewhere.